test/reporting_api_parser_test.dart

```diff
@@ -183,29 +184,68 @@ void main() {
 
       test('a report json string that contains a single row of revenue value',
           () {
-        input =
-            generateInput('Insertion Order ID, Revenue\\n123456,88.88\\n\\n');
+        final reportBody = 'Insertion Order ID, Date, Revenue, Impression\\n'
+            '123456,2020/01/01,88.88,1000\\n';
+        input = generateInput(reportBody);
 
         final actual = ReportingQueryParser.parseRevenueFromJsonString(input);
 
-        final expected = {'123456': '88.88'};
-        expect(actual, expected);
+        expect(actual is Map<String, List<InsertionOrderDailySpend>>, true);
+
+        expect(actual['123456'] is List<InsertionOrderDailySpend>, true);
+        expect(actual['123456'].length, 1);
+        expect(actual['123456'].first.date, DateTime(2020, 1, 1));
+        expect(actual['123456'].first.revenue, '88.88');
+        expect(actual['123456'].first.impression, '1000');
       });
 
       test('a report json string that contains multiple rows of revenue values',
           () {
-        final reportBody = 'Insertion Order ID, Revenue\\n'
-            '111111,88.88\\n222222,999.999\\n333333,0.00\\n\\n';
-        input = generateInput(reportBody);
+        final reportHeader = 'Insertion Order ID, Date, Revenue, Impression\\n';
+        final reportBody = '''
+        111111, 2020/01/01, 100.00, 1000\\n
+        111111, 2020/02/01, 100.00, 1000\\n
+        111111, 2020/03/01, 100.00, 1000\\n
+        222222, 2020/02/01, 200.00, 2000\\n
+        222222, 2020/02/02, 200.00, 2000\\n
+        333333, 2020/03/01, 300.00, 3000\\n
+        ''';
+        input = generateInput(
+            '$reportHeader${reportBody.replaceAll(RegExp(r'\s+'), '')}');
 
         final actual = ReportingQueryParser.parseRevenueFromJsonString(input);
-
-        final expected = {
-          '111111': '88.88',
-          '222222': '999.999',
-          '333333': '0.00'
-        };
-        expect(actual, expected);
+        final firstSpending = actual['111111'];
+        final secondSpending = actual['222222'];
+        final thirdSpending = actual['333333'];
+
+        expect(actual is Map<String, List<InsertionOrderDailySpend>>, true);
+
+        expect(firstSpending is List<InsertionOrderDailySpend>, true);
```

Comment:
you could create your own expected proto and expect equivalence. this would compare all the fields at the same time.

Reply:
I copied the mapEqual function over and make it work for multiMap as well. No longer doing these checks now.